tensorflow_datasets/image/celeba.py

```diff
@@ -117,33 +117,39 @@ def _info(self):
     )
 
   def _split_generators(self, dl_manager):
-    extracted_dirs = dl_manager.download_and_extract({
-        "img_align_celeba": IMG_ALIGNED_DATA,
-        "list_eval_partition": EVAL_LIST,
-        "list_attr_celeba": ATTR_DATA,
-        "landmarks_celeba": LANDMARKS_DATA,
-    })
+    img_align_celeba, list_eval_partition, list_attr_celeba, landmarks_celeba = \
+      dl_manager.download([IMG_ALIGNED_DATA, EVAL_LIST, ATTR_DATA, LANDMARKS_DATA])
+
     return [
         tfds.core.SplitGenerator(
             name=tfds.Split.TRAIN,
             num_shards=10,
             gen_kwargs={
                 "file_id": 0,
-                "extracted_dirs": extracted_dirs,
+                "img_align_celeba": dl_manager.iter_archive(img_align_celeba)
+                "list_eval_partition": dl_manager.iter_archive(list_eval_partition)
+                "list_attr_celeba": dl_manager.iter_archive(list_attr_celeba)
+                "landmarks_celeba": dl_manager.iter_archive(landmarks_celeba)
             }),
         tfds.core.SplitGenerator(
             name=tfds.Split.VALIDATION,
             num_shards=4,
             gen_kwargs={
                 "file_id": 1,
-                "extracted_dirs": extracted_dirs,
+                "img_align_celeba": dl_manager.iter_archive(img_align_celeba)
+                "list_eval_partition": dl_manager.iter_archive(list_eval_partition)
+                "list_attr_celeba": dl_manager.iter_archive(list_attr_celeba)
+                "landmarks_celeba": dl_manager.iter_archive(landmarks_celeba)
             }),
         tfds.core.SplitGenerator(
             name=tfds.Split.TEST,
             num_shards=4,
             gen_kwargs={
                 "file_id": 2,
-                "extracted_dirs": extracted_dirs,
+                "img_align_celeba": dl_manager.iter_archive(img_align_celeba)
```

Comment:
The values from this map are passed as arguments to the _generate_examples - and the code will fail (as you didn't change the arguments in line 183).

Please run the unittests.

Reply:
Yes, @cyfra I am working on that part.
Thanks for the review.